veps/sig-storage/virtiofsd-as-infra.md

```diff
@@ -0,0 +1,175 @@
+strong security boundary by isolating virtiofsd from the host, it introduces
+significant functional limitations:
+* **File Handle support:** Running `virtiofsd` in an unprivileged container
+restricts its ability to track guest files using file handles. This capability
+is crucial for efficient and robust file sharing, especially when dealing with
+PVCs that might have a large number of files. Without proper file handle
+support, `virtiofsd` must rely on file descriptors which are a limited resource.
+
+* **Live Migration challenges**: The inability to properly manage file handles
+directly impacts the safety and reliability of live migration for VMs utilizing
+Virtiofs for PVCs sharing that might be concurrently accessed. During a live
+migration, the `virtiofsd` instance needs to hand over its internal state to the
+target destination. Restrictions on file handles make this hand-off prone to
+issues, potentially leading to data inconsistencies or migration failures.
+
+By moving `virtiofsd` to be managed by `virt-handler` and allowing it to join
+the container's namespaces and cgroups, we aim to overcome these limitations,
+enabling `virtiofsd` file handle functionality while keeping a strong
+security.
+
+## Goals
+
+* Enable `virtiofsd` to utilize file handles, facilitating robust PVC sharing
+with VMs.
+* Facilitate safe and reliable live migration for VMs that leverage Virtiofs
+for PVC sharing.
+* Maintain the rootless execution model for the virtiofs container,
+preserving its security benefits.
+
+## Non Goals
+
+* Using this method to share `configMaps`, `secrets`, `donwardAPIs` or
+`serviceAccounts`; the current implementation of a `virtiofsd` process
+ within a dedicated, unprivileged container will continue serving these volumes.
+
+* Cross cluster live migration and/or storage migration. it is beyond the scope
+of virtiofsd to migrate the contents of the files. In addition, when using file
+handles, both the source and target are required to have access to the same file
+system, since the files are accessed using their inode number.
+
+## Definition of Users
+
+* VM Owners
+* Cluster Admins
+
+## User Stories
+
+* As a KubeVirt user, I want to use Virtiofs for sharing PVCs with my VMs
+without encountering issues related to file descriptor limitations, like
+reaching the open files limit.
+* As a KubeVirt user, I want to be able to live migrate VMs that use Virtiofs
+reliably and safely, ensuring data consistency during migration events.
+* As a KubeVirt administrator, I want to be able to provide robust live
+migration without allowing privileged containers.
+
+## Repos
+
+[KubeVirt](https://github.com/kubevirt/kubevirt)
+
+
+## Design
+
+The management of the `virtiofsd` process will be integrated into the KubeVirt
+infrastructure. The virtiofs container will remain rootless, starting a dummy
+process as PID 1.
+
+The `virt-handler` will launch the `virtiofsd` binary inside the virtiofs
```

Comment:
Could you elaborate a little on that? will there be a separate container under the virt-handler pod, similar to the pr-helper with the dispatcher app? any extra info here would be awesome :pray: